api/store/mongo/tags.go

```diff
@@ -122,11 +102,91 @@ func (s *Store) TagsDelete(ctx context.Context, tenantID string, tag string) (in
 			return int64(0), err
 		}
 
-		return devCount + keyCount + rulCount, nil
+		tagCount, err := s.TagsBulkDeleteTag(sessCtx, tenantID, tag)
+		if err != nil {
+			return int64(0), err
+		}
+
+		return devCount + keyCount + rulCount + tagCount, nil
 	})
 	if err != nil {
 		return int64(0), FromMongoError(err)
 	}
 
 	return count.(int64), nil
 }
+
+func (s *Store) VerifyTagExist(ctx context.Context, tagName, tenant string) (bool, error) {
+	tag := new(models.Tags)
+
+	cursor, err := s.db.Collection("tags").Find(ctx,
+		bson.M{"name": tagName, "tenant_id": tenant})
+	if err != nil {
+		return false, FromMongoError(err)
+	}
+
+	for cursor.Next(ctx) {
+		err := cursor.Decode(tag)
+		if err != nil {
+			return false, err
+		}
+
+		if err != nil {
+			return false, FromMongoError(err)
+		}
+
+		if tag.Name == "" {
+			return false, nil
+		}
+	}
+
+	return true, nil
+}
```

Comment:
Well, you can keep your opinion, but I must say that it's not 'overly complicated', since actually mixing an error with the intended information of a method is not a good pattern/idea (there is a clear separation of what is data, a boolean value, and what is an actual error). indeed, I also will keep the code on this way, if you wish so, since the behavior is the same, and you are just giving an opinion, but again, I'm not criticizing your opinion, just talking about the best interface and logic that fits `VerifyTagExist`.